src/core/path.c

```diff
@@ -518,10 +534,24 @@ static void path_enter_running(Path *p) {
                 return;
         }
 
-        r = manager_add_job(UNIT(p)->manager, JOB_START, trigger, JOB_REPLACE, NULL, &error, NULL);
+        info = activation_event_info_new(UNIT(p));
+        if (!info) {
+                r = -ENOMEM;
+                goto fail;
+        }
+
+        if (free_and_strdup(&(ACTIVATION_EVENT_INFO_PATH(info))->trigger_path_filename, trigger_path) < 0) {
+                r = -ENOMEM;
+                goto fail;
+        }
+
+        r = manager_add_job(UNIT(p)->manager, JOB_START, trigger, JOB_REPLACE, NULL, &error, &job);
         if (r < 0)
                 goto fail;
 
+        if (job)
```

Comment:
the added job might be result of a coalescing event, and thus might already have the `activation_event_info` field set